components/cam/src/physics/crm/sam/ADV_MPDATA/advect_scalar3D.F90

```diff
@@ -109,9 +136,13 @@ subroutine advect_scalar3D (ncrms, f, u, v, w, rho, rhow, flux)
         enddo
       endif
       if (rank.gt.nsubdomains-nsubdomains_x-1) then
+#if defined(_OPENACC)
         !$acc parallel loop collapse(4) async(asyncid)
+#elif defined(_OPENMP)
+        !$omp target teams distribute parallel do collapse(4)
+#endif
         do k=1,nzm
-          do j=ny+1,dimy2_v
```

Comment:
Why change `nyp1` to `ny+1`? Unless `nyp1` is causing a bug, I think we should keep it the same.